core/src/test/java/io/questdb/griffin/AlterTableSetTypeRestartTest.java

```diff
@@ -0,0 +1,384 @@
+                final SqlCompiler compiler = new SqlCompiler(engine)
+        ) {
+            TestUtils.assertSql(
+                    compiler,
+                    context,
+                    "SELECT count() FROM " + tableName,
+                    Misc.getThreadLocalBuilder(),
+                    "count\n" +
+                            count + "\n"
+            );
+        }
+    }
+
+    private static SqlExecutionContext createSqlExecutionContext(CairoEngine engine) {
+        return new SqlExecutionContextImpl(engine, 1).with(
+                AllowAllCairoSecurityContext.INSTANCE,
+                null,
+                null,
+                -1,
+                null
+        );
+    }
+
+    private static void createTable(String tableName, String walMode) throws SQLException {
+        runSqlViaPG("create table " + tableName + " (ts timestamp, x long) timestamp(ts) PARTITION BY DAY " + walMode);
+        LOG.info().$("created table: ").utf8(tableName).$();
+    }
+
+    private static Path doesConvertFileExist(CairoEngine engine, String tableName, boolean doesExist) {
+        final TableToken token = engine.getTableToken(tableName);
+        final Path path = Path.PATH.get().of(engine.getConfiguration().getRoot()).concat(token).concat(WalUtils.CONVERT_FILE_NAME);
+        MatcherAssert.assertThat(Chars.toString(path), Files.exists(path.$()), Matchers.is(doesExist));
+        return doesExist ? path : null;
+    }
+
+    private static void drainWalQueue(CairoEngine engine) {
+        try (final ApplyWal2TableJob walApplyJob = new ApplyWal2TableJob(engine, 1, 1, null)) {
+            walApplyJob.drain(0);
+            new CheckWalTransactionsJob(engine).run(0);
+            // run once again as there might be notifications to handle now
+            walApplyJob.drain(0);
+        }
+    }
+
+    private static void dropTable(String tableName) throws SQLException {
+        runSqlViaPG("drop table " + tableName);
+        LOG.info().$("dropped table: ").utf8(tableName).$();
+    }
+
+    private static void insertInto(String tableName) throws SQLException {
+        runSqlViaPG("insert into " + tableName + " values('2016-01-01T00:00:00.000Z', 1234)");
+        LOG.info().$("inserted 1 row into table: ").utf8(tableName).$();
+    }
+
+    private static void runSqlViaPG(String sql) throws SQLException {
+        try (
+                final Connection connection = DriverManager.getConnection(PG_CONNECTION_URI, PG_CONNECTION_PROPERTIES);
+                final PreparedStatement stmt = connection.prepareStatement(sql)
+        ) {
+            stmt.execute();
+        }
+    }
+
+    private static void setType(String tableName, String walMode) throws SQLException {
+        runSqlViaPG("alter table " + tableName + " set type " + walMode);
+        LOG.info().$("scheduled table type conversion for table ").utf8(tableName).$(" to ").utf8(walMode).$();
```

Comment:
wal mode can be logged with .$(), there is not utf8 in it.

Reply:
fixed in [509c5d5](https://github.com/questdb/questdb/pull/2947/commits/509c5d5677a0aa8f5bac45ee8878683fc16321fe)